diff --color -ruN a/js/update-slots.js b/js/update-slots.js
--- a/js/update-slots.js	2025-06-04 15:05:33
+++ b/js/update-slots.js	2025-06-04 15:03:46
@@ -31,20 +31,22 @@
 
   Drupal.behaviors.vaa_update_slots = {
     attach: function (context, settings) {
-      $(once('vaa_init', '.vaa-slots > div', context)).each(function () {
-        const $slot = $(this);
-        const $parent = $slot.parent().parent().closest('fieldset');
+      if($(context).find('select[name$="[display_id]"]').length > 0 && $(context).find('select[name$="[display_id]"]').val()) {
+        $(once('vaa_init', '.vaa-slots > div', context)).each(function () {
+          const $slot = $(this);
+          const $parent = $slot.parent().parent().closest('fieldset');
 
-        const $loading = $('<img class="vaa-loading" src="' + settings.path.baseUrl + settings.viewfield_argument_helper.path + '/images/loading.gif">');
-        $loading.hide();
-        $slot.append($loading);
+          const $loading = $('<img class="vaa-loading" src="' + settings.path.baseUrl + settings.viewfield_argument_helper.path + '/images/loading.gif">');
+          $loading.hide();
+          $slot.append($loading);
 
-        updateSlots($parent, $slot);
-
-        $parent.find('select').on('change', function(e) {
           updateSlots($parent, $slot);
-        })
-      });
+
+          $parent.find('select').on('change', function(e) {
+            updateSlots($parent, $slot);
+          })
+        });
+      }
     }
   };
 })(jQuery);
\ No newline at end of file
