diff --color -ruN a/frontback.module b/frontback.module
--- a/frontback.module	2026-03-05 17:17:23
+++ b/frontback.module	2026-03-05 17:17:12
@@ -72,7 +72,9 @@
  */
 function _frontback_page_match($pages) {
   // Logic from core RequestPath condition plugin.
-  $pages = mb_strtolower($pages);
+  if(isset($pages)) {
+    $pages = mb_strtolower($pages);
+  }
   if (!$pages) {
     return FALSE;
   }
