diff --color -ruN a/src/Commands/MailgunSanitizeCommands.php b/src/Commands/MailgunSanitizeCommands.php
--- a/src/Commands/MailgunSanitizeCommands.php	2025-07-30 16:57:53
+++ b/src/Commands/MailgunSanitizeCommands.php	2025-07-30 16:58:38
@@ -5,7 +5,7 @@
 use Consolidation\AnnotatedCommand\CommandData;
 use Drupal\Core\Queue\QueueFactory;
 use Drush\Commands\DrushCommands;
-use Drush\Commands\sql\\SanitizePluginInterface;
+use Drush\Drupal\Commands\sql\SanitizePluginInterface;
 use Symfony\Component\Console\Input\InputInterface;
 
 /**
